universal-application-tool-0.0.1/app/controllers/dev/AzuriteFileStorageStrategy.java

```diff
@@ -26,9 +26,17 @@ public Result create(StoredFileRepository storedFileRepository, Request request)
     return redirect(routes.FileUploadController.index().url()).flashing("success", successMessage);
   }
 
-  private void updateFileRecord(StoredFileRepository storedFileRepository, String key) {
+  private void updateFileRecord(
+      StoredFileRepository storedFileRepository, String key, String userFileName) {
     StoredFile storedFile = new StoredFile();
     storedFile.setName(key);
+    storedFile.setUserFileName(getPrefixedUserFileName(key, userFileName));
     storedFileRepository.insert(storedFile);
   }
+
+  private String getPrefixedUserFileName(String fileName, String userFileName) {
+    String[] newFileName = fileName.split("/");
```

Comment:
What would an example of bad file name input be? having trouble wrapping my head around what to check for. The userFileName is just the name of the file that's uploaded. Maybe I should rename it to "original file name?"

Reply:
update: I renamed it. originalFileName makes more sense

Reply:
like if the filename is somehow 'no-slahes.jpg' or something?

Reply:
In that case, `split` should just return a string with a list size of one, right? we're assuming all the files will be prefixed with the slashes, but if they're not, the whole name would just get replaced. I think since the file name is generated programmatically instead of from user input, we shouldn't run into any badly formatted strings.

Reply:
cool that sounds good to me